src/typescript/frontend/src/components/OrderBook.tsx

```diff
@@ -0,0 +1,175 @@
+    <div
+      // key={key}
+      className={`relative my-[1px] flex min-h-[25px] min-w-full items-center justify-between text-xs `}
+    >
+      <div
+        className={`z-10 ml-4 text-right ${
+          type === "sell" ? "text-red-400" : "text-green"
+        }`}
+      >
+        {order.price}
+      </div>
+      <div className="z-10 mr-4 text-white">{order.size}</div>
+      <div
+        className={`absolute right-0 z-0 h-full opacity-30 ${
+          type === "sell" ? "bg-red-400" : "bg-green"
+        }`}
+        // dynamic taillwind?
+        style={{ width: `${(100 * order.size) / highestSize}%` }}
+      ></div>
+    </div>
+  );
+
+  return (
+    <div className="divider-solid flex grow flex-col divide-y divide-neutral-600">
+      {/* title row */}
+      <div className={"mx-4 my-[12px]"}>
+        <div className={"flex justify-between"}>
+          <p className={"font-jost text-white"}>Order Book</p>
+          {/* select */}
+          <Listbox value={precision} onChange={setPrecision}>
+            <div className="relative z-30  min-h-[30px] border border-neutral-600 py-[8px] pl-[11px] pr-[8px]">
+              <Listbox.Button className=" flex min-w-[4em] justify-between font-roboto-mono text-neutral-300">
+                {precision}
+                <ChevronDownIcon className="my-auto ml-1 h-5 w-5 text-neutral-500" />
+              </Listbox.Button>
+              <Listbox.Options className="absolute mt-2 w-full bg-black shadow ring-1 ring-neutral-500">
+                {PrecisionOptions.map((precision, i) => (
+                  <Listbox.Option
+                    key={i}
+                    value={precision}
+                    className="weight-300 cursor-pointer px-4 py-1 font-roboto-mono text-xs text-neutral-300 hover:bg-neutral-800"
+                  >
+                    {precision}
+                  </Listbox.Option>
+                ))}
+              </Listbox.Options>
+            </div>
+          </Listbox>
+        </div>
+        <div
+          className={`mt-[11px] flex justify-between text-xs text-neutral-500`}
+        >
+          <div className={``}>PRICE (USD)</div>
+          <div>AMOUNT (APT)</div>
+        </div>
+      </div>
+      {/* bids ask spread scrollable container */}
+      <div className={`scrollbar grow overflow-y-auto`}>
+        {/* i don't understand why this suddenly solved my problem but it did lol */}
+        {/* the problem i was having was that the rows were making a flex parent increase in size,
+        which is not expected. the goal was to have the rows scrollable with a dynamically sized parent
+        according to window size but not have the parent increase in size when the rows were added.
+        */}
+        {/* basically my understanding of what's going on is:
+        since the rows are already overflowing the child they aren't affecting the size of this 'new' parent?
+         */}
+        <div className="max-h-0">
```

Comment:
I feel like using relative for the parent, and absolute for the child feels more natural for the overflow element? (I’ll be honest, I have no clue what `max-h-0` does to make things work there)

Reply:
I wonder if it’s possible to set the initial position to the middle purely using CSS (I spent a bunch of time trying to figure this out, but setting `top-1/2 -translate-1/2` doesn’t let you scroll all the way to the top so I’m not sure haha)

Reply:
looks like coinbase does 1 of 2 things:
1. have skeleton elements for each row that are autopopulated by orderbook data
2. a loading overlay that covers the orderbook underneath